### PR TITLE
DEV-837

### PR DESCRIPTION
**Pull request by tranlyvu**
update error handling in join_project_existing_user method (adapter-deterlab)

### CODE REVIEW

**Comment by tranlyvu** [comment]
Chunwang, please leave comment here if you change anything, don't need to verbally explain to me. Christ always leave his comments here

**Comment by ZHANGChunwang** [comment]
1. I didn't merge 2 functions. Previously there were 2 function names, join_project_logged_on and join_project_existing_user. Join_project_logged_on didn't do anything, it just simply calls join_project_existing_user. So i just removed it.
2. "???" is the error message I need you to find out. That's why I want to talk to you this morning.
3. Please be clear of what messages are shown to the users and what messages are just for passing info to the SIO. You should keep the messages passing between adapter and SIO as short, standard and consistent as possible. If possible, can just return "ok" to sio.

**Comment by tranlyvu** [comment]
2) In both situation when existing user re-apply for approved or non-approved team, the return output is "You are already a member"

File: adapter.py [hunk 1]
```diff
@@ -189,12 +189,44 @@ def POST(self):
...
+		logging.info('Join project existing user: response from Deter -- %s', out)
+
+		if "The project leader has been notified of your application" in out :
+			logging.info('Join project existing user: ok')
+			return '{"msg":"join project existing user ok"}'
+		elif "Invalid Project Name" in out :
+			logging.warning('Join project existing user: error, project %s does not exist', pid)
+			return '{"msg":"project not found on DeterLab"}'
+		elif "You are already a member" in out :
+			logging.warning('Join project existing user: error, user %s is either an approved or non-approved member of project %s', uid, pid)
```
Comment (ZHANGChunwang) [inline adapter.py]: change 'non-approved' to 'pending'

File: adapter.py [hunk 2]
```diff
@@ -189,12 +189,44 @@ def POST(self):
...
+			return '{"msg":"user is either an approved or non-approved member on DeterLab"}'
```
Comment (ZHANGChunwang) [inline adapter.py]: return '{"msg":"you are already a member of the project on DeterLab"}'

**Comment by ZHANGChunwang** [comment]
Good enough to merge, but need to run local tests to test the 4 if-else branches :)

**Pull request merged**